scripts/tools/generate_esp32_chip_factory_bin.py

```diff
@@ -277,15 +277,23 @@ def gen_spake2p_params(passcode):
 
 
 def populate_factory_data(args, spake2p_params):
-    FACTORY_DATA['discriminator']['value'] = args.discriminator
-    FACTORY_DATA['iteration-count']['value'] = spake2p_params['Iteration Count']
-    FACTORY_DATA['salt']['value'] = spake2p_params['Salt']
-    FACTORY_DATA['verifier']['value'] = spake2p_params['Verifier']
-    FACTORY_DATA['dac-cert']['value'] = os.path.abspath(args.dac_cert)
-    FACTORY_DATA['pai-cert']['value'] = os.path.abspath(args.pai_cert)
-    FACTORY_DATA['cert-dclrn']['value'] = os.path.abspath(args.cd)
-    FACTORY_DATA['dac-key']['value'] = os.path.abspath('dac_raw_privkey.bin')
-    FACTORY_DATA['dac-pub-key']['value'] = os.path.abspath('dac_raw_pubkey.bin')
+    if args.discriminator is not None:
+        FACTORY_DATA['discriminator']['value'] = args.discriminator
+
+    if spake2p_params is not None:
```

Comment:
do we need None check for non-numeric values? I can see that args.discriminator could be 0 however for the rest, the `if foo:` feels more pythonic than `if foo is not None` since the value is not boolean/numeric bu a path/pbject instead.

```python
 if spake2p_params:
    # ....
 
if args.dac_cert:
    FACTORY_DATA['dac-cert']['value'] = os.path.abspath(args.dac_cert)
if args.pai_cert:
    FACTORY_DATA['pai-cert']['value'] = os.path.abspath(args.pai_cert)
# ...
```

Reply:
Thanks @andy31415, I removed these and all the occurrences of `is not None`.